packages/open-next/src/core/routing/routeMatcher.ts

```diff
@@ -0,0 +1,64 @@
+import { AppPathRoutesManifest, RoutesManifest } from "config/index";
+import type { RouteDefinition } from "types/next-types";
+import type { RouteType } from "types/open-next";
+
+// Add the locale prefix to the regex so we correctly match the rawPath
+const optionalLocalePrefixRegex = RoutesManifest.locales.length
+  ? `^/(?:${RoutesManifest.locales.map((locale) => `${locale}/?`).join("|")})?`
+  : "^/";
```

Comment:
Nit: the following should work:

```suggestion
const optionalLocalePrefixRegex = `^/(?:${RoutesManifest.locales.map((locale) => `${locale}/?`).join("|")})?`;
```